neuro_san/http_sidecar/http_server_app.py

```diff
@@ -13,21 +13,124 @@
 See class comment for details
 """
 from typing import Any, Dict
+import time
+from threading import Lock, Thread
+
 from tornado.web import Application
+from tornado.ioloop import IOLoop
+
+from neuro_san.interfaces.event_loop_logger import EventLoopLogger
 
 
 class HttpServerApp(Application):
     """
     Class provides customized Tornado application for neuro-san service -
     with redefined internal logger so we can include custom request metadata.
     """
-    def __init__(self, handlers):
+    # pylint: disable=too-many-instance-attributes
+    SHUTDOWN_TIMEOUT_SECONDS: int = 30
+
+    def __init__(self, handlers, requests_limit: int, logger: EventLoopLogger):
         """
         Constructor:
         :param handlers: list of request handlers
         """
         # Call the base constructor
         super().__init__(handlers=handlers)
+        self.total: int = 0
+        self.num_processing: int = 0
+        self.requests_stats: Dict[str, int] = {}
+        self.requests_limit: int = requests_limit
+        self.logger: EventLoopLogger = logger
+        self.serving: bool = True
+        self.shutdown_initiated: bool = False
+        self.lock: Lock = Lock()
+        self.shutdown_thread = None
+
+    def is_serving(self) -> bool:
```

Comment:
Extend HttpServerApp (which is essentially our server proper) with all properties to handle requests statistics collecting and logging.
We also need things for server shutdown when number of executed requests exceeds the limit.